src/iexamine.cpp

```diff
@@ -3980,7 +3980,20 @@ void smoker_finalize(player &, const tripoint &examp)
             product = "dry_fish";
         } else if( item_it.typeId() == "raw_beans" ) {
             product = "dry_beans";
-        } else if( item_it.typeId() == "sweet_fruit" || item_it.typeId() == "coconut" || item_it.typeId() == "can_coconut" ) {
+        } else if( item_it.typeId() == "sweet_fruit" ||
```

Comment:
You can use some flag instead of using hardcoded item ids, so people could add more items via json and they would be smokable too.

Reply:
Yes, that's what I meant in the "Additional context" section. This PR is the quickest solution that copies state of surrounding code. JSONizing looks more like a new feature and not a bugfix as I see it. I'm not sure I will have time to JSONize it right now.

Reply:
I'm looking into it right now. Can you please point me to some commit with a good example of JSONizing?

Reply:
Stuck on this https://github.com/int-ua/Cataclysm-DDA/commit/6bcdc2c27ed144ae3131bed337a9f0c639dee423#diff-3196f85e7ab533f3659c29eedb38d09cR3976 , what's wrong with this line? It crashes the game.

Reply:
Looks like there is a smoke also in items and `item_it.type->comestible->smoking_result` failing because it doesn't have `comestible`